__tests__/src/services/ClientService.test.ts

```diff
@@ -5,28 +5,60 @@
  *
  */
 
-import { ServiceCategory, clientService } from '../../../src/services/ClientService';
+import { AppConstants } from 'src/constants';
+import { AppConstantType } from '@types';
+import { clientService } from '../../../src/services/ClientService';
 import supabaseClient from '../../../utils/supabaseClient';
 
-describe('ClientService', () => {
-  it('should get service categories', async () => {
-    const response = { data: [new ServiceCategory({})], error: null };
-    const mockQueryBuilder = {
-      select: jest.fn(() => Promise.resolve(response)),
+// Idea for mock from https://stackoverflow.com/questions/77411385/how-to-mock-supabase-api-select-requests-in-nodejs
+jest.mock('@supabase/supabase-js', () => ({
+  /** Need to use non-arrow function to bind setTestData function to returned 'supabaseClient' object  */
+  // eslint-disable-next-line prefer-arrow-callback
+  createClient: jest.fn().mockImplementation(function mockCreateClient() {
```

Comment:
Oh interesting. Is this similar the the dependency injection we talked about before?

Reply:
Yes exactly. The supabase module isn't even loaded during this test, jest will intercept all imports to it (by modules loaded during the tests is this file) and substitute the mock module I provide as the second argument